src/java.base/share/classes/sun/security/x509/X509CertImpl.java

```diff
@@ -739,7 +739,7 @@ public void delete(String name)
         id = attr.getPrefix();
 
         if (id.equalsIgnoreCase(INFO)) {
-            if (attr.getSuffix() != null) {
+            if (!(attr.getSuffix() != null)) {
```

Comment:
> The test can be much simpler than the program included in the bug report. Something like this:
> 
>     1. Create an empty `X509CertImpl`
> 
>     2. Add "x509.info"
> 
>     3. Add "x509.info.issuer"
> 
>     4. Remove "x509.info.issuer"
> 
>     5. Add "x509.info.issuer" again
> 
> 
> Without the fix, step 4 removes the whole info and step 5 would fail.
I am not a getting a clear idea about how to add issuer and delete etc. 
Could you please point me to some example?